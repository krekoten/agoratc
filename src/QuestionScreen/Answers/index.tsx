import React from "react"
import type { Answer } from "../../types"

import AnswerItem from "./AnswerItem"
import { RadioGroup } from "@mui/material"

interface AnswersParams {
  answers: Answer[]
}

function Answers({ answers }: AnswersParams): JSX.Element {
  return <RadioGroup data-testid="answers" className="options">
    {answers.map(({ text, correct }, index) => <AnswerItem text={text} correct={correct} key={index} />)}
  </RadioGroup>
}

export default Answers
