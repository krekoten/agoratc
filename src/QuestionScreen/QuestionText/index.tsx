import React from "react"

import { Typography } from "@mui/material"

function QuestionText({ text }: { text: string }): JSX.Element {
  return (
    <Typography variant="h5" className="question">
      {text}
    </Typography>
  )
}

export default QuestionText
