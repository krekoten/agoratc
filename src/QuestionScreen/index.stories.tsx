import React from "react"

import type { Meta, StoryObj } from "@storybook/react"

import QuestionScreen from "."

const meta: Meta<typeof QuestionScreen> = {
  component: QuestionScreen,
}

export default meta

type Story = StoryObj<typeof QuestionScreen>

const question = {
  text: "some question",
  answers: [
    { text: "Answer 1", correct: false },
    { text: "Answer 2", correct: true },
  ],
}

const onCorrectAnswer = () => {}
const onIncorrectAnswer = () => {}

export const Layout: Story = {
  render: () => (
    <QuestionScreen
      question={question}
      onCorrectAnswer={onCorrectAnswer}
      onIncorrectAnswer={onIncorrectAnswer}
    />
  ),
}
