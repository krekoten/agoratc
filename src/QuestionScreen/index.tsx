import React, { useState } from "react"

import { Box, Button } from "@mui/material"

import QuestionText from "./QuestionText"
import Answers from "./Answers"

import type { Question } from "../types"

interface QuestionParams {
  question: Question
  onCorrectAnswer: () => void
  onIncorrectAnswer: () => void
}

function QuestionScreen({
  question: { text, answers },
  onCorrectAnswer,
  onIncorrectAnswer,
}: QuestionParams): JSX.Element {
  const [option, setOption] = useState<number | null>(null)

  function onClick(): void {
    if (option !== null && answers[option].correct) {
      onCorrectAnswer()
    } else {
      onIncorrectAnswer()
    }
    setOption(null)
  }

  return (
    <Box sx={{ display: "flex", flexDirection: "column", alignItems: "left", margin: 2 }}>
      <QuestionText text={text} />
      <Answers
        answers={answers}
        value={option}
        onChange={(value: string) => {
          setOption(Number(value))
        }}
      />
      <Button className="next" variant="contained" color="primary" onClick={onClick}>
        Next
      </Button>
    </Box>
  )
}

export default QuestionScreen
