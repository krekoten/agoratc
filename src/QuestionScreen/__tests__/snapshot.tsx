import React from "react"
import renderer from "react-test-renderer"

import QuestionScreen from ".."

test("QuestionScreen Snapshot", () => {
  const questionText = "some question"
  const answers = [
    { text: "Answer 1", correct: false },
    { text: "Answer 2", correct: true },
  ]
  const question = {
    text: questionText,
    answers,
  }
  const tree = renderer
    .create(
      <QuestionScreen
        question={question}
        onCorrectAnswer={() => {}}
        onIncorrectAnswer={() => {}}
      />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
