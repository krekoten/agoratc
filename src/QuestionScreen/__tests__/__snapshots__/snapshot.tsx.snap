// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionScreen Snapshot 1`] = `
<div
  className="MuiBox-root css-1wbuatv"
>
  <h5
    className="MuiTypography-root MuiTypography-h5 question css-ag7rrr-MuiTypography-root"
  >
    some question
  </h5>
  <div
    className="MuiFormGroup-root options css-dmmspl-MuiFormGroup-root"
    data-testid="answers"
    role="radiogroup"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="Answer 1"
          onChange={[Function]}
          required={false}
          type="radio"
          value={0}
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Answer 1
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          name="Answer 2"
          onChange={[Function]}
          required={false}
          type="radio"
          value={1}
        />
        <span
          className="css-hyxlzm"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
            data-testid="RadioButtonCheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
            />
          </svg>
        </span>
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Answer 2
      </span>
    </label>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium next css-sghohy-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Next
  </button>
</div>
`;
